Mage.Sets/src/mage/cards/n/Necromentia.java

```diff
@@ -0,0 +1,122 @@
+package mage.cards.n;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.ChooseACardNameEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.predicate.mageobject.NamePredicate;
+import mage.game.Game;
+import mage.game.permanent.token.Token;
+import mage.game.permanent.token.ZombieToken;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInLibrary;
+import mage.target.common.TargetOpponent;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class Necromentia extends CardImpl {
+
+    public Necromentia(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{1}{B}{B}");
+
+        // Choose a card name other than a basic land card name. Search target opponent's graveyard, hand, and library for any number of cards with that name and exile them. That player shuffles their library, then creates a 2/2 black Zombie creature token for each card exiled from their hand this way.
+        this.getSpellAbility().addEffect(
+                new ChooseACardNameEffect(ChooseACardNameEffect.TypeOfName.NONBASIC_LAND_NAME).setText("Choose a card name other than a basic land card name")
```

Comment:
Wrong setting. Must be `NOT_BASIC_LAND_NAME`. That's why you don't need custom text here.

Wrong/custom text is smell of wrong effect/ability implementation. Use `showCardInfo()` unit test to view card's auto-generated text.

Reply:
oh, damn, rookie mistake there.. I even tested with non-basic lands specifically and didn't verify other card types 🤦... fixed